Sources/Bot/Services/DispatchService.swift

```diff
@@ -0,0 +1,198 @@
+            .startWithValues { pullRequests in
+                self.dispatchInitial(pullRequests: pullRequests)
+            }
+
+        gitHubEvents.events
+            .observe(on: scheduler)
+            .observeValues { [weak self] gitHubEvent in
+                switch gitHubEvent {
+                case let .pullRequest(event):
+                    self?.pullRequestDidChange(event: event)
+                case let .status(event):
+                    self?.statusChecksDidChange(event: event)
+                case .ping:
+                    break
+                }
+            }
+    }
+
+    private func dispatchInitial(pullRequests: [PullRequest]) {
+        let dispatchTable = Dictionary(grouping: pullRequests) { pullRequest in pullRequest.target.ref }
+        for (branch, pullRequestsForBranch) in dispatchTable {
+            makeMergeService(targetBranch: branch, scheduler: self.scheduler, initialPullRequests: pullRequestsForBranch)
+        }
+    }
+
+    private func pullRequestDidChange(event: PullRequestEvent) {
+        logger.log("📣 Pull Request did change \(event.pullRequestMetadata) with action `\(event.action)`")
+        let targetBranch = event.pullRequestMetadata.reference.target.ref
+        if let service = mergeServices[targetBranch] {
+            service.pullRequestChangesObserver.send(value: (event.pullRequestMetadata, event.action))
+        } else {
+            makeMergeService(targetBranch: targetBranch, scheduler: self.scheduler, initialPullRequests: [event.pullRequestMetadata.reference])
+        }
+    }
+
+    private func statusChecksDidChange(event: StatusEvent) {
+        // No way to know which MergeService this event is supposed to be for – isRelative(toBranch:) only checks for head branch not target so not useful here
+        // So we're sending it to all MergeServices, and they'll filter them themselves based on their own queues
+        for mergeServiceForBranch in mergeServices.values {
+            mergeServiceForBranch.statusChecksCompletionObserver.send(value: event)
+        }
+    }
+
+    @discardableResult
+    private func makeMergeService(targetBranch: String, scheduler: DateScheduler, initialPullRequests: [PullRequest] = []) -> MergeService {
+        let mergeService = MergeService(
+            targetBranch: targetBranch,
+            integrationLabel: integrationLabel,
+            topPriorityLabels: topPriorityLabels,
+            requiresAllStatusChecks: requiresAllStatusChecks,
+            statusChecksTimeout: statusChecksTimeout,
+            initialPullRequests: initialPullRequests,
+            logger: logger,
+            gitHubAPI: gitHubAPI,
+            scheduler: scheduler
+        )
+        self.mergeServices[targetBranch] = mergeService
+        self.healthcheck.startMonitoring(mergeServiceHealthcheck: mergeService.healthcheck)
+        mergeServiceLifecycleObserver?.send(value: .created(mergeService))
+        mergeService.state.producer
```

Comment:
this observation as well maybe? it's just a bit weird that we do everything here, return the value, but then never use it (as far as I can tell)

we can then extract this observations into method, like `observerMergeServiceLifetime` or something like that